File: src/main/java/cs/java/event/CSEventRegistrations.java
package cs.java.event;

import cs.java.collections.CSList;
import cs.java.event.CSEvent.CSEventRegistration;

import static cs.java.lang.CSLang.YES;
import static cs.java.lang.CSLang.list;

/**
 * Created by renetik on 26/12/17.
 */

public class CSEventRegistrations {

    private CSList<CSEventRegistration> _registrations = list();
    private boolean _active = YES;

    public CSEventRegistrations() {
    }

    public CSEventRegistrations(CSEventRegistration... registrations) {
        _registrations.append(registrations);
    }

    public void cancel() {
        for (CSEventRegistration reg : _registrations) reg.cancel();
        _registrations.clear();
    }

    public CSEventRegistrations addAll(CSEventRegistration... registrations) {
        _registrations.append(registrations);
        return this;
    }

    public CSEventRegistration add(CSEventRegistration registration) {
        registration.setActive(_active);
        return _registrations.put(registration);
    }

    public void setActive(boolean active) {
        _active = active;
        for (CSEventRegistration registration : _registrations) registration.setActive(_active);
    }
}

File: library/src/main/java/renetik/android/java/event/CSEventRegistrations.kt
package renetik.android.java.event

import renetik.android.java.collections.list
import renetik.android.java.event.CSEvent.CSEventRegistration
import renetik.android.lang.CSLang.YES

class CSEventRegistrations() {

    private val registrations = list<CSEventRegistration>()
    private var active = YES

    constructor(vararg registrations: CSEventRegistration) : this() {
        this.registrations.append(*registrations)
    }

    fun cancel() {
        for (reg in registrations) reg.cancel()
        registrations.clear()
    }

    fun addAll(vararg registrations: CSEventRegistration): CSEventRegistrations {
        this.registrations.append(*registrations)
        return this
    }

    fun add(registration: CSEventRegistration): CSEventRegistration {
        registration.isActive = active
        return registrations.put(registration)
    }

    fun setActive(active: Boolean) {
        this.active = active
        for (registration in registrations) registration.isActive = active
    }
}
